crates/holochain_p2p/src/spawn.rs

```diff
@@ -56,10 +56,30 @@ impl Default for HolochainP2pConfig {
     }
 }
 
+/// See [NetworkCompatParams::proto_ver].
+pub const HCP2P_PROTO_VER: u32 = 2;
+
 /// Some parameters used as part of a protocol compability check during tx5 preflight
-#[derive(Debug, Default, serde::Serialize, serde::Deserialize)]
+#[derive(
+    Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, serde::Serialize, serde::Deserialize,
+)]
 pub struct NetworkCompatParams {
+    /// The current protocol version. This should be incremented whenever
+    /// any breaking protocol changes are made to prevent incompatible
+    /// nodes from talking to each other.
+    pub proto_ver: u32,
```

Comment:
It'd be great to be able to dump out a connectivity rules report of some kind. It's so often something we end up debugging. Just a thought for later, but having something that tells you whether a pair of conductors should be able to connect based on their version, network settings and DNA hashes etc, would be a useful debugging tool.

Reply:
At the very least, we can trace it out until we come up with something better: https://github.com/holochain/holochain/pull/4783/commits/c2ceeeb4063e4d7bb3dbf1972828921dba11d9b1